File: web/src/utils/consumeSpaces.ts
import isSpace from "../utils/isSpace";

export default function consumeSpaces(text: string, i: number): string {
	let result = "";
	for (; i < text.length; i++) {
		let char = text[i];
		if (isSpace(text.charCodeAt(i))) {
			result += char;
		} else {
			break;
		}
	}
	return result;
}

File: dotnet/Allmark/Utils/ConsumeSpaces.cs
namespace Allmark;

public static partial class Utils
{
	public static string ConsumeSpaces(string text, int i)
	{
		var result = "";
		for (; i < text.Length; i++)
		{
			char c = text[i];
			if (IsSpace((int)c))
			{
				result += c;
			}
			else
			{
				break;
			}
		}
		return result;
	}
}
